### PR TITLE
code review

### PR DESCRIPTION
**Pull request by sam3ay**

### CODE REVIEW

File: scripts/dictquery.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    """Fetches data from nested dictionary
+    Args:
+        input_dict: nested Dictionary
+        listofpath: path of desired value
+    Returns:
+        list of values at path specified by input list
+    """
+    output_list = []
+    for path in listofpath:
+        output_list.append(functools.reduce(dict.get, path.split('/'), input_dict))
```
Comment (Damien-Black) [inline scripts/dictquery.py]: Nice use of `functools.reduce`

Doesn't this keep making `dict.get` calls if the call fails early on in path? example:
```python
dictquery({}, ['rekt/gg/no/re'])
```
will make some calls, then fail when dict,get reruns `None` and None is attempted to be used again.

File: scripts/build_meta.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from dict_extract import *
+from dictquery import *
+from xmldictconv import *
```
[the remark this answers is not in the document]
Comment (sam3ay) [reply scripts/build_meta.py]: Agreed, modularization artifact will remove.

File: scripts/build_meta.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        input_dict: optional variable, dictionary, contains metadata information
+    Returns:
+        list of strings, metadata information
+    Notes:
+        'DESIGN/SAMPLE_DESCRIPTOR/PRIMARY_ID',
+        'DESIGN/LIBRARY_DESCRIPTOR/LIBRARY_NAME,
+        'PLATFORM/ILLUMINA/INSTRUMENTAL_MODEL', ILLUMINA, SC
+    """
+    # retrieve parent directory from path and specify experiment xml file
+    exper_dict = input_dict
```
[the remark this answers is not in the document]
Comment (sam3ay) [reply scripts/build_meta.py]: Ahh, true.  I do seem to remember that we're not supposed to alter function inputs directly, so maybe I'm remembering incorrectly or I have to make a deep copy.

File: scripts/build_meta.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    exp_loc = os.path.abspath(os.path.join(path, os.pardir))
+    exp_id = os.path.split(exp_loc)[1]
+    exp_name = exp_loc + os.sep + exp_id + '.experiment.xml'
```
[the remark this answers is not in the document]
Comment (sam3ay) [reply scripts/build_meta.py]: Given that input, the output should look like path='a/b/c/c.xml'

File: scripts/build_meta.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    try:
+        experiment = next(gen_dict_extract(identifier, exper_dict[exp_name]), None)
+    except KeyError:
+        exper_dict[exp_name] = xmldictconv(exp_name)
+        experiment = next(gen_dict_extract(identifier, exper_dict[exp_name]), None)
```
[the remark this answers is not in the document]
Comment (sam3ay) [reply scripts/build_meta.py]: True, though I think the last line could just be
`exper_dict[exp_name] = experiment`

File: scripts/dict_extract.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+def gen_dict_extract(value, var, des_dict=None):
```
[the remark this answers is not in the document]
Comment (sam3ay) [reply scripts/dict_extract.py]: The overall idea here is to pull a dictionary[A] out of a list of dictionaries(which contain nested dictionaries[B] with the value of interest).  The value is only used to identify the dictionary[A] in the list to extract, not the nested dictionaries[B] within that dictionary[A].  I'll try to make that coherent and add it to the docstrings.

File: scripts/dictquery.py (hunk 1 above)
Comment (sam3ay) [reply scripts/dictquery.py]: I didn't test the fail state...  I'll write a test and figure that out.  Still, that's probably what'll happen.  A couple unnecessary dict.get calls.

File: scripts/build_meta.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Damien-Black) [reply scripts/build_meta.py]: yeah thats correct

File: scripts/build_meta.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sam3ay) [reply scripts/build_meta.py]: Use string .format() to create the filename in the exp_name line

**Pull request merged**